src/client/unittests/common/baseTestManager.ts

```diff
@@ -144,14 +171,15 @@ export abstract class BaseTestManager {
         }
 
         this._status = TestStatus.Running;
-        this.createCancellationToken();
+        this.stop();
+        this.createCancellationToken(CancellationTokenType.testDicovery);
         // If running failed tests, then don't clear the previously build UnitTests
         // If we do so, then we end up re-discovering the unit tests and clearing previously cached list of failed tests
         // Similarly, if running a specific test or test file, don't clear the cache (possible tests have some state information retained)
         const clearDiscoveredTestCache = runFailedTests || moreInfo.Run_Specific_File || moreInfo.Run_Specific_Class || moreInfo.Run_Specific_Function ? false : true;
-        return this.discoverTests(clearDiscoveredTestCache, true)
+        return this.discoverTests(clearDiscoveredTestCache, true, true)
```

Comment:
Geez do I miss keyword arguments.

Reply:
Agreed, the plan is to remove all of these arguments and go with an object (easier).
In some places I got rid of them and added enums (slightly better, improves readability).
E.g. some of the functions have a number of arguments (specially the new changes, but with DI this will hopefully change to a single service provider interface)